### PR TITLE
move from use_lvm flag on guided fs things to more generic capabilities

### PR DESCRIPTION
**Pull request by mwhudson**
We should absolutely not land this anywhere that will be picked up by ubuntu-desktop-installer for 23.04 partly because this is an API change and partly because this area of the code is not the best tested! Still I'd like @jpnurmi to see this as they start to think about the canary installer.

### CODE REVIEW

**Comment by mwhudson** [comment]
Hm the disadvantage of capabilities being a list like this is that there is no scope to indicate why a capability might be unavailable. Maybe it should be a list of some slightly richer data structure...

**Comment by jpnurmi** [comment]
> We should absolutely not land this anywhere that will be picked up by ubuntu-desktop-installer for 23.04

We've changed the desktop installer to track ubuntu/lunar:

- https://github.com/canonical/ubuntu-desktop-installer/pull/1580

**Review by dbungert — APPROVED** [review]

File: subiquity/common/types.py [hunk 1]
```diff
@@ -397,6 +403,7 @@ class GuidedResizeValues:
 @attr.s(auto_attribs=True)
 class GuidedStorageTargetReformat:
     disk_id: str
+    capabilities: List[GuidedCapability]
```
[the remark this answers is not in the document]
Comment (mwhudson) [reply subiquity/common/types.py:406]: I'm not sure I understand. Having these fields be mandatory certainly makes the test changes noisy but if the client just takes a target and posts it back to the server then it doesn't matter if they are optional or not? Or do you want to make it easier for clients to construct GuidedStorageTarget by hand? (The server does already entirely ignore the capabilities fields  in v2_guided_POST)

[the remark this answers is not in the document]
Comment (dbungert) [reply subiquity/common/types.py:406]: As a general API design thing, I find it strange that the ability to just take an object like one of these Targets and post it right back means that is has some fields that will never make sense to manually construct.  This is a general observation and not specific to this change.  Perhaps I just need to let that one go.

For certain this isn't a problem I expect to fix in this PR.

**Comment by mwhudson** [comment]
I've set up a new 'canary-23.04' branch and am going to merge this branch into it. More to come!

**Pull request merged**